### PR TITLE
fix(payment methods): use new payment method eligibity endpoint

### PR DESCRIPTION
**Pull request by schnogz**
## Description
- Fix to ensure non-gold users are shown payment methods for Buy/Sell instead of the currency unsupported screen
- Removes duplicate entries of same APIs.  The two removed where not used in the codebase

### CODE REVIEW

**Review by blockdylanb — COMMENTED** [review]
Looks good to me. Works locally as well. I'll let Milan take a look and approve if he doesn't have any objections.

**Review by milan-bc — APPROVED** [review]
Tested and code reviewed LGTM!

**Pull request merged**
